thing-url-adapter.js

```diff
@@ -145,21 +150,24 @@ class ThingURLDevice extends Device {
   }
 
   closeWebsocket() {
-    if (this.ws) {
-      this.ws.removeAllListeners('close');
-      this.ws.removeAllListeners('error');
+    if (this.ws !== null) {
+      this.closingWs = true;
 
       if (this.ws.readyState === WebSocket.OPEN) {
         this.ws.close();
       }
 
-      this.ws = null;
+      // Allow the cleanup code in createWebsocket to handle shutdown
```

Comment:
The cleanupAndReopen code should be updated to null `this.ws` then for paranoia's sake.

Reply:
Fair point. Done.

Reply:
:+1: